core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -831,9 +832,11 @@ private[spark] class Executor(
     }
 
     val message = Heartbeat(executorId, accumUpdates.toArray, env.blockManager.blockManagerId)
+    val heartbeatIntervalInSec =
+      conf.getTimeAsMs("spark.executor.heartbeatInterval", "10s").millis.toSeconds.seconds
```

Comment:
After discussing with @gatorsmile , we found there is a bug here. If `spark.executor.heartbeatInterval` is less than one second, it will always be 0 and timeout. (https://github.com/scala/scala/blob/v2.11.12/src/library/scala/concurrent/impl/Promise.scala#L209)

This may break some user's tests that set a small timeout.

Reply:
@dbtsai Could we document it in the release note?

Reply:
@ajithme @srowen We need to fix this ASAP.

Reply:
I see your point, but that isn't new behavior. This was always parsed as 'seconds' here before, so anything less than a second would have resulted in 0. It's a separate bug but does sound like a problem

Reply:
Sorry. I was not clear. I meant, for example, if `spark.executor.heartbeatInterval` is `900` without a time unit, it will be converted to 0 now.

Reply:
Yeah I agree that this is a closely-related bug and fix; the `master` change fixed both but this change just fixes the unit inconsistency, not also the truncation of this value to seconds.

Release notes probably can't hurt but I am not clear a setting of < "1000" would have ever even worked in practice.

Reply:
I'll put it in the release note. Thanks!

Reply:
Release note added, http://spark.apache.org/releases/spark-release-2-4-1.html

Reply:
Opened https://github.com/apache/spark/pull/24329 to fix the issue